library/src/main/kotlin/com/livefront/gsonkotlinadapter/util/KParameter.kt

```diff
@@ -3,8 +3,30 @@ package com.livefront.gsonkotlinadapter.util
 import com.google.gson.annotations.SerializedName
 import java.lang.reflect.Field
 import java.lang.reflect.Modifier
+import kotlin.reflect.KClass
 import kotlin.reflect.KParameter
 
+/**
+ * Returns the default value of the primitive this [KParameter] represents or `null` if it does not
+ * represents primitive.
+ */
+internal val KParameter.defaultPrimitiveValue: Any?
+    get() {
+        if (type.isMarkedNullable) return null
+        return when (type.classifier as? KClass<*>) {
+            null -> null
```

Comment:
Is this not covered in the `else` below?

Reply:
Just being explicit... I can remove it.